src/ui/public/embeddable/embeddable.ts

```diff
@@ -53,34 +44,17 @@ export interface EmbeddableMetadata {
   editUrl?: string;
 }
 
-interface EmbeddableOptions {
-  metadata?: EmbeddableMetadata;
-  render?: (domNode: HTMLElement, containerState: ContainerState) => void;
-  destroy?: () => void;
-  onContainerStateChanged?: (containerState: ContainerState) => void;
-}
-
 export abstract class Embeddable {
   public readonly metadata: EmbeddableMetadata = {};
 
   // TODO: Make title and editUrl required and move out of options parameter.
-  constructor(options: EmbeddableOptions = {}) {
-    this.metadata = options.metadata || {};
-
-    if (options.render) {
-      this.render = options.render;
-    }
-
-    if (options.destroy) {
-      this.destroy = options.destroy;
-    }
-
-    if (options.onContainerStateChanged) {
-      this.onContainerStateChanged = options.onContainerStateChanged;
-    }
+  constructor(metadata: EmbeddableMetadata = {}) {
```

Comment:
I changed the interface here.  Felt like allowing all those functions to be passed in was more confusing than requiring passing in a super class, and it was only used in the disabled lab embeddable, and it also conflicted with making this an abstract class, doing it that way.